frontend/src/views/submissions.js

```diff
@@ -58,55 +63,36 @@ class Submissions extends Component {
 
   // makes an api call to upload the selected file
   uploadData = () => {
-    // 'upload_status_cookie' in Cookies is used to communicate between the functions in api.js and those in submissions.js. It lets us keep track of the upload process for submissions, and all the http requests involved. (Note that this is different than a submission's compile_status in the database.)
-    // A value of 0 indicates that the submission is still in progress.
-    // When a submission finishes, api.js changes this value to something else.
-    Cookies.set("upload_status_cookie", 10);
-    // The upload_status state is used internally by this component.
-    // (Currently, it mirrors upload_status_cookie, but is part of state to make working with React easier.)
-    this.setState({ upload_status: 10 });
-
-    // Concurrent upload processes can be problematic; we've made the decision to disable concurrency.
-    // This is achieved by refreshing the submission upload components, which have buttons disabled while upload_status is 0.
-    this.renderHelperSubmissionForm();
-    this.renderHelperSubmissionStatus();
-
-    Api.newSubmission(this.state.selectedFile, null);
-
-    // The method in api.js will change Cookies' upload_status_cookie during the process of an upload.
-    // To check changes, we poll periodically.
-    this.interval = setInterval(() => {
-      let upload_status_cookie_value = Cookies.get("upload_status_cookie");
-      if (upload_status_cookie_value != 10) {
-        // Submission process terminated (see api.js).
-
-        // refresh the submission status, for use on this component
-        this.setState({ upload_status: upload_status_cookie_value });
-
-        // refresh the submission button, etc, to allow for a new submission
-        this.renderHelperSubmissionForm();
-        this.renderHelperSubmissionStatus();
-
-        // refresh team submission tables, to display the submission that just occured
-        Api.getTeamSubmissions(this.gotSubmissions);
-        this.renderHelperCurrentTable();
-        this.renderHelperLastTable();
-
-        // Done waiting for changes to upload_status_cookie, so stop polling.
-        clearInterval(this.interval);
+    this.setState({ upload_status: "loading" });
+    Api.newSubmission(
+      this.state.selectedFile,
+      this.state.package,
+      this.state.description,
+      this.props.episode,
+      (success) => {
+        if (success) {
+          this.setState({ upload_status: "success", selectedFile: null });
```

Comment:
Perhaps, ideally also blank out the rest of the form, in addition to just the file. That way it actually looks like you reset the form, instead of having half of it disappear.